hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -923,12 +937,123 @@ public <N extends Number> SqmExpression<N> abs(Expression<N> x) {
 		);
 	}
 
+	@Override
+	public JpaExpression<Duration> duration(long magnitude, TemporalUnit unit) {
+		return new SqmToDuration<>(
+				literal( magnitude ),
+				new SqmDurationUnit<>( unit, getLongType(), this ),
+				getTypeConfiguration().standardBasicTypeForJavaType( Duration.class ),
+				this
+		);
+	}
+
+	@Override
+	public JpaExpression<Long> magnitude(TemporalUnit unit, Expression<Duration> duration) {
+		return new SqmByUnit(
+				new SqmDurationUnit<>( unit, getLongType(), this ),
+				(SqmExpression<Duration>) duration,
+				getLongType(),
+				this
+		);
+	}
+
+	@Override
+	public JpaExpression<Duration> add(Expression<Duration> x, Expression<Duration> y) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.ADD,
+				(SqmExpression<Duration>) x, (SqmExpression<Duration>) y );
+	}
+
+	@Override
+	public JpaExpression<Duration> add(Expression<Duration> x, Duration y) {
+		return createSqmArithmeticNode( BinaryArithmeticOperator.ADD,
+				(SqmExpression<Duration>) x, value( y ) );
+	}
+
+	@Override
+	public JpaExpression<Duration> subtract(Expression<Duration> x, Expression<Duration> y) {
```

Comment:
Analogously to the above, how about `durationSubtract(Expression<Duration> x, Expression<Duration> y)` and `temporalSubtract(Expression<? extends Temporal> x, Expression<Duration> y)`?